hack/check-files-alignment.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash -e
+
+#
+# Copyright 2019 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+source hack/project_info.sh
+roles_yaml_path=config/rbac/role.yaml
+crd_yaml_path=config/crd/bases/csi.ibm.com_ibmblockcsis.yaml
+
+check_generation (){
+  echo "check generation"
+  project_dirname=ibm-block-csi-operator
+  cd ..
+  cp -r $project_dirname ./$project_dirname-expected
+  cd $project_dirname-expected/
+  make update
+  cd ..
+  diff -qr --exclude=bin $project_dirname $project_dirname-expected/
+  rm -rf $project_dirname-expected/
+  cd $project_dirname
+}
+
+verify_yaml_doc_in_multidoc_yaml(){
+  yamls_dictionary=${1#*=}
+  eval "declare -A yaml_kinds_by_origin_yaml_path="${yamls_dictionary}
+  for orignial_yaml in ${!yaml_kinds_by_origin_yaml_path[@]}; do
+    export resource_type=${yaml_kinds_by_origin_yaml_path[${orignial_yaml}]}
+    diff <(yq e '... comments=""' $orignial_yaml) <(yq eval '(. | select(.kind == env(resource_type)))' $operator_yaml_path)
+  done
+}
+
+verify_full_operator_yaml_is_aligned(){
+  echo "check full operator yaml alignment"
+  declare -A yaml_kinds_by_origin_yaml_path=(
+      [$roles_yaml_path]="ClusterRole"
+      [$crd_yaml_path]="CustomResourceDefinition"
+      ["config/rbac/service_account.yaml"]="ServiceAccount"
+      ["config/rbac/role_binding.yaml"]="ClusterRoleBinding"
+      ["config/manager/manager.yaml"]="Deployment"
```

Comment:
why do we need the kinds hardcoded? can't we find them dynamically?

Reply:
I will open a ticket

Reply:
to add `resource_type=$(yq e .kind $original_yaml)`?